sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -38,6 +39,40 @@ import (
 // Example 2: 777777777777.dkr.ecr.cn-north-1.amazonaws.com.cn/my_image:latest
 var ecrRegex = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?.*`)
 
+// Used to define valid container types
+type ContainerType string
+
+const (
+	Host      ContainerType = "host"
+	Bootstrap ContainerType = "bootstrap"
+)
+
+// Returns an error if an invalid type is used
+func (ct ContainerType) IsValid() error {
+	switch ct {
+	case Host, Bootstrap:
+		return nil
+	}
+
+	return errors.New("Invalid container type")
+}
+
+// Returns the persistent base directory for the container type
+func (ct ContainerType) PersistentDir() (string, error) {
+  // Return an error to prevent successful calls on invalid
+  // container types
+  dir, err := "", errors.New("Invalid container type")
+
+	switch ct {
+	case Host:
+		dir, err = "host-containers", nil
+	case Bootstrap:
+		dir, err ="bootstrap-containers", nil
+	}
+
+  return dir, err
+}
```

Comment:
As above, I wouldn't use an `error` here.

```suggestion
// PersistentDir returns the persistent base directory for the container type
func (ct ContainerType) PersistentDir() (string, bool) {
	switch ct {
	case Host:
		return "host-containers", true
	case Bootstrap:
		return "bootstrap-containers", true
	}
	return "", false
}
```